### PR TITLE
Simple ADTs with a single constructor are records

### PR DESCRIPTION
**Pull request by Halbaroth**
This PR fixes a bug in `D_cnf` that have been exposed by #1130. Before merging `Enum` into `ADT` the situation was as follows:
- simple ADT with several constructors and with at least one payload was sent to the `Adt` theory
- simple ADT with several constructors but no payload was sent to `Enum` theory
- simple ADT with a single constructor was sent to the `Record` theory.

The last corner case includes a corner corner case: an ADT with a single constructor without payload (so basically the unit type with an extra step...)

The `D_cnf` included a bug. It produced single constructors without payload of type ADT instead of type record but the check done in `is_mine_symb` prevented from sending it to `Adt`. As I removed this check, we got an assert on the following input:
```smt2
(set-logic ALL)
(declare-datatype t ((Record)))
(define-fun a () t Record)
(check-sat)
```

After this patch, the situation is as follows:
- simple ADT with several constructors is sent to the `Adt` theory
- simple ADT with only one constructor is sent to the `Record` theory
- mutually recursive ADT whose all the types are record is sent to the `Record` theory

### CODE REVIEW

**Review by hra687261 — APPROVED** [review]

File: src/lib/frontend/d_cnf.ml [hunk 1]
```diff
@@ -975,8 +975,10 @@ let rec mk_expr
                 E.mk_record [] ty
               | Tadt _ as ty ->
                 E.mk_constr (Uid.of_dolmen tcst) [] ty
-              | _ ->
-                assert false
+              | Tunit ->
+                E.void
+              | ty ->
+                Fmt.failwith "unexpected type %a@." Ty.print ty
```
Comment (hra687261) [inline src/lib/frontend/d_cnf.ml:981]: Is there a test that shows that Tunit is used? And that we crash if we don't match it here?

Comment (Halbaroth) [reply src/lib/frontend/d_cnf.ml:981]: There are 2000 regressions on `ae-format` without this case, so yes, we have tests :)
Actually the builtin `unit` type of `Dolmen` is an ADT but a distinct builtin type `Tunit` in Alt-Ergo. It's a corner corner corner case...

I tested this PR on `ae-format` and now there is no regression anymore ;)

Comment (Halbaroth) [reply src/lib/frontend/d_cnf.ml:981]: I will add a unit test to catch unit (sic) case

Comment (Halbaroth) [reply src/lib/frontend/d_cnf.ml:981]: Finally, I removed the `Ty.Tunit` type of AE in #1148, so this corner case disappears.
